### PR TITLE
Update person class to become abstract

### PR DESCRIPTION
**Pull request by kohkaijie**
Update person class to become abstract and added two new classes, Gender and Ic, which are now fields in the person class. 

I have yet to make the methods in the Person class abstract, which should be done next as the Doctor and Patient classes will have different toString and isEqual methods.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -1,11 +1,7 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (cmHuang777) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:4]: I think we should not wildcards for import statements.

**Review by cmHuang777 — COMMENTED** [review]
I noticed that some of the import statements are using wildcard, could you change them back to not using wildcard?

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -14,9 +21,12 @@
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.person.Address;
 import seedu.address.model.person.Email;
+import seedu.address.model.person.Gender;
+import seedu.address.model.person.Ic;
```
Comment (tanveersingh10) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:25]: small nit but should we make this NRIC to be consistent with the CliSyntax.java? anything tbh

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -9,10 +9,7 @@
 import seedu.address.commons.core.index.Index;
 import seedu.address.commons.util.StringUtil;
 import seedu.address.logic.parser.exceptions.ParseException;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Phone;
+import seedu.address.model.person.*;
```
Comment (tanveersingh10) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:12]: shouldn't wildcard import

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -95,6 +92,36 @@ public static Email parseEmail(String email) throws ParseException {
...
 
+    /**
+     * Parses a {@code String ic} into an {@code Ic}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code ic} is invalid.
+     */
+    public static Ic parseIc(String ic) throws ParseException {
+        requireNonNull(ic);
+        String trimmedIc = ic.trim();
```
Comment (tanveersingh10) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:103]: add .toUpperCase() also in case the user enters a lowecase ic number for consistency

**Review by tanveersingh10 — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java
Comment (Mohammed-Faizzzz) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:1]: This File is missing a blank line at EOF, causing the build to fail. Take note and add it in.

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 2 above)
Comment (Mohammed-Faizzzz) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java:25]: I agree with Tanveer

**Review by Mohammed-Faizzzz — APPROVED** [review]
Merging to Pull and Edit.

**Pull request merged**